third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -80,6 +80,8 @@ pub(crate) struct ExpTranslator<'env, 'translator, 'module_translator> {
     pub placeholder_map: BTreeMap<NodeId, ExpPlaceholder>,
     /// A cached unification context, containing information about structs in the context.
     pub unification_context: CachedUnificationContext,
+    /// A flag to indicate whether in the middle of translating lhs part of an expression
+    pub translate_lhs: bool,
```

Comment:
Actually I am wondering whether we need this flag because if there is an operation of freeze to be inserted at LHS, it may already indicate an error?

Reply:
Maybe an example like this?
```
let x = vector[1, 2, 3];
let r = &mut x;
*r[*r[1]] = 4;
```
Not sure about Move vs Rust syntax.

Reply:
For this example, since Move treats vector::borrow as function call and the insertion of freeze operation for function  arguments is not controlled by this flag. However, it actually reveals another bug. Thanks @brmataptos!

Reply:
Updated the name and use of this flag.